risc0/zkvm/src/guest/env.rs

```diff
@@ -71,6 +94,132 @@ pub fn syscall(syscall: SyscallName, to_host: &[u8], from_host: &mut [u32]) -> s
     }
 }
 
+// TODO(victor) Check these and any other doc links.
+/// Error encountered during a call to [crate::verify]
+///
+/// Note that an error is only returned for "provable" errors. In particular, if
+/// the host fails to find a receipt matching the requested image_id and
+/// journal, this is not a provable error. In this case, the [crate::verify]
+/// will not return.
+#[derive(Debug)]
+pub enum VerifyError {}
+
+impl fmt::Display for VerifyError {
+    fn fmt(&self, _f: &mut fmt::Formatter) -> fmt::Result {
+        unreachable!("VerifyError error enum is empty and cannot be constructed");
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for VerifyError {}
+
+/// Verify there exists a receipt for an execution with the given `image_id` and
+/// `journal`.
+///
+/// In order to be valid, the [Receipt] must have have `ExitCode::Halted(0)`, an
+/// empty assumptions list, and an all-zeroes input hash. It may have any post
+/// [SystemState].
+pub fn verify(image_id: Digest, journal: &[u8]) -> Result<(), VerifyError> {
+    let journal_digest: Digest = Sha256::digest(journal).as_slice().try_into().unwrap();
```

Comment:
Changed this to avoid a potential panic

Reply:
Generally I think we should try to avoid `expect` in the guest because it's very costly. Maybe this is a bad idea, but I haven't found much use for `expect` generally since I can get the line of code the panic occurs and the Error type itself should contain information about the error.

Reply:
It is actually a good practice to always use `.expect("Proof why this never ever happens")` instead of `.unwrap()` (except in tests where crashes are fine). It significantly helps with auditability and generally quality of the software, especially in blockchain context. And with careful use of type system most of these should compile away.

`.expect()` is not to print a better message to developer than `.unwrap()` when things crash, it is to prove to the reader that code will not crash under any circumstances (or in rare cases that it might, but explain why it is acceptable), otherwise error handling must be used instead.